### ISSUE TITLE
dart2js: find a way not to add all checked mode helpers in resolution queue

### ISSUE THREAD

**Issue by rakudrama** [dart-lang/sdk#13155]
We register all the checked mode helpers in resolution because the check, 

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;emitter.nativeEmitter.requiresNativeIsCheck(element);

does not work property until after resolution.

This work item is to find a way to avoid enqueueing all 22 helpers.

The problem is that selecting the correct helper now depends on whether a
mixin is used on some native class or a subclass of a native class. You can't
tell from the type in your hand (the mixin or a subtype thereof) that a mixin is
used in another class until that other class has been resolved.  Resolving that
other class depends on allocations or native calls in methods that have not been
resolved yet.  So the selection of the correct helper depends the set of all
resolved types.

Basically, we need to avoid subtype queries until after resolution is complete.

I'm putting this at low priority because:

(1) I believe the selection criteria for helpers should be based of the static type of the receiver, and not the type of the test.  This information is not known at resolution time.

(2) No extra code is generated by putting a few more top level methods in the resolution queue.
